src/components/ui/OakTextInput/OakTextInput.tsx

```diff
@@ -0,0 +1,196 @@
+   */
+  readOnly?: boolean;
+  /**
+   * Sets the value. Use this in controlled components;
+   */
+  value?: string;
+  /**
+   * Sets the initial value. Use this for an uncontrolled component;
+   */
+  defaultValue?: string;
+  /**
+   * Used to target the input element in tests.
+   */
+  "data-testid"?: string;
+  placeholder?: string;
+  onChange?: ChangeEventHandler<HTMLInputElement>;
+  /**
+   * Alters the appearance of the input field to indicate whether the input is valid or invalid.
+   */
+  validity?: "valid" | "invalid";
+  /**
+   * Displays an icon at the start of the input.
+   */
+  startEnhancerIconName?: OakIconName;
+  /**
+   * Displays an icon at the end of the input.
+   */
+  endEnhancerIconName?: OakIconName;
+  $iconColor?: OakCombinedColorToken;
+  $validBorderColor?: OakCombinedColorToken;
+  $invalidBorderColor?: OakCombinedColorToken;
+  $validIconColor?: OakCombinedColorToken;
+  $invalidIconColor?: OakCombinedColorToken;
+} & Partial<StyledTextInputWrapperProps> &
+  Pick<InternalTextInputProps, "$width" | "$maxWidth" | "onInitialFocus">;
+
+const StyledTextInputWrapper = styled(OakFlex)<StyledTextInputWrapperProps>`
+  &:hover {
+    cursor: text;
+  }
+
+  &:focus-within:not(:has(input:disabled, input:read-only)) {
+    box-shadow: ${(props) =>
+      props.$focusRingDropShadows
+        .map((dropShadow) => parseDropShadow(dropShadow))
+        .join(",")};
+  }
+
+  background: ${(props) => parseColor(props.$background)};
+
+  @media (hover: hover) {
+    &:hover:not(:focus-within, :has(input:read-only)) {
+      background: ${(props) => parseColor(props.$hoverBackground)};
+    }
+  }
+
+  &:has(input:read-only) {
+    border-color: ${(props) => parseColor(props.$readOnlyBorderColor)};
+    color: ${(props) => parseColor(props.$readOnlyColor)};
+  }
+
+  &:has(input:disabled) {
+    background: ${(props) => parseColor(props.$disabledBackgroundColor)};
+    color: ${(props) => parseColor(props.$disabledColor)};
+  }
+`;
+
+/**
+ * Default input which can be extended to create specialised inputs.
+ */
+export const OakTextInput = ({
+  type = "text",
+  $borderColor = "border-primary",
+  $readOnlyBorderColor = "border-neutral",
+  $focusRingDropShadows = [
+    "drop-shadow-centered-yellow",
+    "drop-shadow-centered-grey",
+  ],
+  $background = "bg-primary",
+  $hoverBackground = "bg-neutral",
+  $disabledBackgroundColor = "bg-neutral",
+  $color = "text-primary",
+  $disabledColor = "text-disabled",
+  $readOnlyColor = "text-subdued",
+  validity,
+  $iconColor = "icon-inverted",
+  $validBorderColor = "border-success",
+  $invalidBorderColor = "border-error",
+  $validIconColor = "icon-success",
+  $invalidIconColor = "border-error",
+  startEnhancerIconName,
+  endEnhancerIconName,
+  ...props
+}: OakTextInputProps) => {
+  let borderColor: OakCombinedColorToken;
+  let iconColor: OakCombinedColorToken;
+  let readOnlyBorderColor: OakCombinedColorToken;
+
+  switch (validity) {
```

Comment:
I'm in two minds as to whether the validation styles warrant another flavour of input component.

On the one hand it is "logic", but validation seems intrinsic to a form field so perhaps it belongs at this level since the colours can still be set through style props.

My take away from what i've seen in the library so far is that the intention is for components to become more opinionated through each layer of composition.

So perhaps this does belong here, but `OakQuizInput` is where we apply an end enhancer icon of tick/cross and validity `'valid' | 'invalid'`when the feedback is `'correct' | 'incorrect'`



`base/InternalTextInput` => unstyled, fully transparent but with the ability to set dimensions and spacing. Intended for consumption by a bonafide component ready for use
`ui/OakTextInput` => some opinionated branded styling with strong defaults and baked in states. Allows any colour to be changed through styling props
`integrated/OakQuizInput` =>  very opinionated, pretty much single use.

🤔 thinking about it, does `OakQuizInput` even have a place in a component library. Is it so specific that it actually belongs in the app?

Reply:
I don't have any opinions on your last paragraph (all I care about here is that the base components be independently testable and well-documented), but I agree with the assumptions in the rest of your comment

Reply:
On the last para there's something of a fuzzy line but the rule I'm applying is that if the component is "dumb" ie. still doesn't require knowledge of context then it can live in this library. The ultimate aim here is that we have components which can be used in multiple apps. A quiz checkbox could certainly be one of those.